src/sequence.lisp

```diff
@@ -196,31 +196,39 @@
 (defun copy-seq (sequence)
   (subseq sequence 0))
 
-
-;;; Reduce (based on SBCL's version)
-
-(defun reduce (function sequence &key key from-end (start 0) end (initial-value nil ivp))
+(defun %reduce (function sequence key index from-end step-fn compare-fn end value)
```

Comment:
Tests are always good, go ahead.

If running the tests become to slow, we can always create suites so we don't run the full test suite every time.